### PR TITLE
Infoset view

### LINKED ISSUE #35: Infoset diff view
Add a view that accepts a diff event and displays the content.

### LINKED ISSUE #36: Request infoset dump
Ability to obtain dump of current infoset at any point during the parse.

### PR DESCRIPTION
**Pull request by jw3**
Adds a virtual doc view of the infoset.  In addition to adding the obvious function of viewing this infoset, this work is also attempting to demonstrate an alternative way of managing the editor views by splitting the update out from the display.

### commands
- display: displays the view based on the temp file
- save:  writes the current infoset state to a specified file
  - closes #36
- diff: displays a diff of the current infoset state against the previous infoset state
  - closes #35

### CODE REVIEW

File: src/infosetView.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            onDidChange = this.onDidChangeEmitter.event;
+
+            provideTextDocumentContent(uri: vscode.Uri): string {
+                return fs.readFileSync(uri.path).toString();
+            }
+        };
+        context.subscriptions.push(vscode.workspace.registerTextDocumentContentProvider('daffodil:infoset', myProvider));
+
+        context.subscriptions.push(vscode.commands.registerCommand('infoset.update', async (e: daf.InfosetEvent) => {
+            let path = '/tmp/infoset.xml';
```
Comment (jw3) [inline src/infosetView.ts:27]: Will track this file with some type of unique id for the debug session.

Comment (jw3) [reply src/infosetView.ts]: Use the `session.id`

**Comment by jw3** [comment]
Getting close, needs portable temp files yet.

**Review by Shanedell — APPROVED** [review]

**Comment by jw3** [comment]
Diffs here are provided by keeping tabs on the previous infoset state and then using the built in vs code diff tool to run it against the current state.  Pretty simple, but effective.

image

Can diff at any point with ctrl+p diff command.

There are some interesting implications of diffing things from more than one step back as well, but didnt implement a history of any sort.

**Comment by jw3** [comment]
This diff-of-previous-state is similar to how the cli debugger works for the diff command.

**Pull request merged**